### PR TITLE
Stop ignoring some warnings that we really shouldn't be.

### PR DESCRIPTION
**Pull request by asherkin**
alliedmodders/sourcepawn#50 needs to land before this.

@psychonic

### CODE REVIEW

**Comment by psychonic** [comment]
I'm not crazy about -Wswitch. Could we change it to just warn?

Looks like there are some other things that need to be tweaked anyway to fix the build.

**Comment by psychonic** [comment]
I just rebased this against master. Still not a fan of Wswitch, but if it compiles, 🚢

**Comment by Headline** [comment]
Just waiting on alliedmodders/metamod-source#52 and then we should *hopefully* be good to go

File: core/logic/smn_core.cpp [hunk 1]
```diff
@@ -951,7 +951,7 @@ static cell_t LogStackTrace(IPluginContext *pContext, const cell_t *params)
 
 	g_Logger.LogError("[SM] Stack trace requested: %s", buffer);
 	g_Logger.LogError("[SM] Called from: %s", pPlugin->GetFilename());
-	for (size_t i = 0; i < arr.length(); i)
+	for (size_t i = 0; i < arr.length(); ++i)
```
Comment (asherkin) [inline core/logic/smn_core.cpp]: Well that is a suspicious bug to be fixed here, might be best spinning that out to its own PR as it is a behaviour change?

File: extensions/sdktools/clientnatives.h [hunk 2]
```diff
@@ -1,37 +0,0 @@
-/**
```
Comment (asherkin) [inline extensions/sdktools/clientnatives.h]: What happened here?

File: pushbuild.txt [hunk 3]
```diff
@@ -1,75 +1,76 @@
...
+Beeeeeeeees.
+Beads.
+snakesssssssss; maple
+R.I.P. Wade Boggs
+Your tier1 tower is under attack..
+Christmas!!!
+the mousse angles are smooth
+spoilers
+oh key you're so good to me
+well, ok rüd
```
Comment (KyleSanderson) [inline pushbuild.txt]: god damn it... GH merge conflict rewrote all the line endings... :|

File: extensions/sqlite/AMBuilder [hunk 4]
```diff
@@ -19,6 +19,7 @@ for arch in SM.archs:
   ]
   if builder.target.platform == 'linux':
     binary.compiler.postlink += ['-ldl', '-lpthread']
+    binary.compiler.cflags += ['-Wno-error=unused-const-variable']
```
Comment (KyleSanderson) [inline extensions/sqlite/AMBuilder]: This is not an option on gcc-4.9

Comment (KyleSanderson) [reply extensions/sqlite/AMBuilder]: Correction, GCC < 6 doesn't have this option.

Comment (Headline) [reply extensions/sqlite/AMBuilder]: Should just require a check for applicable versions I suppose, I presume this warning didn’t exist on gcc before then?

Comment (KyleSanderson) [reply extensions/sqlite/AMBuilder]: That's right, check out Travis-CI for compiler compliance.

**Comment by peace-maker** [comment]
This requires permissive warnings due to hl2sdk code, so it's not as effective as it could be without touching all hl2sdk branches. Even though this is still desired, someone could take a new chance at this using newer compilers.
